source/flow.hpp

```diff
@@ -22,8 +23,8 @@ class Flow {
     void update();
 
 private:
-    Device *m_src;
-    Device *m_dest;
+    IRoutingDevice *m_src;
+    IRoutingDevice *m_dest;
```

Comment:
It seems like these interfaces were meant here:
```suggestion
    ISender *m_src;
    IReceiver *m_dest;
```

Reply:
Yes, I are right